### PR TITLE
login using identity server (electron)

### PR DESCRIPTION
**Pull request by 5ebastianMeier**
Depends on:
https://github.com/process-engine/bpmn-studio/pull/573
https://github.com/process-engine/skeleton/pull/63

## What did you change?

This PR adds support for the same behavior described in the PR linked above for electron usage.

You can now run the test cases described in the other PR using the regular BPMN-Studio build.

### Setup
* Clone https://github.com/process-engine/identity_server and run the server as described in the `README.md`
* Open `http://localhost:5000` in the browser
   * make sure under Clients > BPMN-Studio > Application URLs the following entries are set:
      * bpmn-studio://signin-oidc - [x] Callback - [ ] Signout
      * bpmn-studio://signout-oidc - [x] Callback - [x] Signout
* Clone this feature branch of BPMN-Studio and run `npm install && npm run electron-build-macos`
* Navigate to `dist/mac/` and run the application 

_Note: for everything to work out of the box, the Identity Server has to run on `localhost:5000`._

### Test
See https://github.com/process-engine/bpmn-studio/pull/573.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [ ] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: config/development/process_engine/flow_node_instance_repository.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "password": null,
+  "database": null,
+  "host": null,
+  "port": null,
+  "dialect": "sqlite",
+  "storage": "flow_node_instance.sqlite",
+  "supportBigNumbers": true,
+  "resetPasswordRequestTimeToLive": 12,
+  "logging": false
+}
```
Comment (SteffenKn) [inline config/development/process_engine/flow_node_instance_repository.json]: Add Newline

File: config/development/process_engine/process_model_repository.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "password": null,
+  "database": null,
+  "host": null,
+  "port": null,
+  "dialect": "sqlite",
+  "storage": "process_model.sqlite",
+  "supportBigNumbers": true,
+  "resetPasswordRequestTimeToLive": 12,
+  "logging": false
+}
```
Comment (SteffenKn) [inline config/development/process_engine/process_model_repository.json]: Add Newline

File: src/modules/authentication/authentication.service.ts [hunk 3]
```diff
@@ -7,22 +7,22 @@ import {
...
-  ILogoutResult,
 } from '../../contracts/index';
 
 const LOCAL_STORAGE_TOKEN_KEY: string = 'process-engine-token';
 
 @inject(EventAggregator, 'AuthenticationRepository', 'TokenRepository')
 export class AuthenticationService implements IAuthenticationService {
 
-  public tokenRepository: ITokenRepository;
+  private eventAggregator: EventAggregator;
```
Comment (SteffenKn) [inline src/modules/authentication/authentication.service.ts]: private => `_eventAggregator`

File: src/modules/authentication/authentication.service.ts [hunk 4]
```diff
@@ -7,22 +7,22 @@ import {
...
+  private authenticationRepository: IAuthenticationRepository;
```
Comment (SteffenKn) [inline src/modules/authentication/authentication.service.ts]: private => `_authenticationRepository`

File: src/modules/authentication/authentication.service.ts [hunk 5]
```diff
@@ -7,22 +7,22 @@ import {
...
+  private tokenRepository: ITokenRepository;
```
Comment (SteffenKn) [inline src/modules/authentication/authentication.service.ts]: private => `_tokenRepository`

**Comment by Vyperus** [comment]
Electron version seems to work at first sight. After discussion @5ebastianMeier will now:

- Add a differentiation between the browser and electron environment to configure the openid-client and everything aorund it correctly.
- Merge the base branch to allow further testing.

File: src/app.ts [hunk 6]
```diff
@@ -1,18 +1,53 @@
-import {inject} from 'aurelia-framework';
-import {OpenIdConnect} from 'aurelia-open-id-connect';
-import {RouterConfiguration} from 'aurelia-router';
+import { inject } from 'aurelia-framework';
+import { OpenIdConnect } from 'aurelia-open-id-connect';
+import {Router, RouterConfiguration} from 'aurelia-router';
+import { NewAuthenticationService } from './modules/authentication/new_authentication.service';
```
Comment (Vyperus) [inline src/app.ts]: Delete Whitespaces

File: src/modules/authentication/new_authentication.service.ts [hunk 7]
```diff
@@ -1,12 +1,15 @@
 import {EventAggregator} from 'aurelia-event-aggregator';
-import {inject} from 'aurelia-framework';
-import {OpenIdConnect} from 'aurelia-open-id-connect';
-import {User} from 'oidc-client';
+import { inject } from 'aurelia-framework';
```
Comment (Vyperus) [inline src/modules/authentication/new_authentication.service.ts]: Delete whitespaces.

**Pull request merged**